Source/WebCore/css/typedom/transform/CSSTranslate.cpp

```diff
@@ -57,6 +60,55 @@ ExceptionOr<Ref<CSSTranslate>> CSSTranslate::create(Ref<CSSNumericValue> x, Ref<
     return adoptRef(*new CSSTranslate(is2D, WTFMove(x), WTFMove(y), z.releaseNonNull()));
 }
 
+ExceptionOr<Ref<CSSTranslate>> CSSTranslate::create(CSSFunctionValue& cssFunctionValue)
+{
+    auto makeTranslate = [&](const Function<ExceptionOr<Ref<CSSTranslate>>(Vector<Ref<CSSNumericValue>>)>& create) -> ExceptionOr<Ref<CSSTranslate>> {
+        Vector<Ref<CSSNumericValue>> components;
+        for (auto componentCSSValue : cssFunctionValue) {
+            auto valueOrException = CSSStyleValueFactory::reifyValue(componentCSSValue);
+            if (valueOrException.hasException())
+                return valueOrException.releaseException();
+            if (!is<CSSNumericValue>(valueOrException.returnValue()))
+                return Exception { TypeError, "Expected a CSSNumericValue."_s };
+            components.append(downcast<CSSNumericValue>(valueOrException.releaseReturnValue().get()));
+        }
+        return create(components);
+    };
+
+    switch (cssFunctionValue.name()) {
+    case CSSValueTranslateX:
+        return makeTranslate([](Vector<Ref<CSSNumericValue>> components) {
+            ASSERT(components.size() == 1);
+            return CSSTranslate::create(components[0], CSSNumericFactory::px(0), nullptr);
+        });
+    case CSSValueTranslateY:
+        return makeTranslate([](Vector<Ref<CSSNumericValue>> components) {
+            ASSERT(components.size() == 1);
+            return CSSTranslate::create(CSSNumericFactory::px(0), components[0], nullptr);
+        });
+    case CSSValueTranslateZ:
+        return makeTranslate([](Vector<Ref<CSSNumericValue>> components) {
+            ASSERT(components.size() == 1);
+            return CSSTranslate::create(CSSNumericFactory::px(0), CSSNumericFactory::px(0), components[0].ptr());
+        });
+    case CSSValueTranslate:
+        return makeTranslate([](Vector<Ref<CSSNumericValue>> components) {
+            ASSERT(components.size() == 1 || components.size() == 2);
+            if (components.size() == 2)
+                return CSSTranslate::create(components[0], components[1], nullptr);
+            return CSSTranslate::create(components[0], CSSNumericFactory::px(0), nullptr);
+        });
+    case CSSValueTranslate3d:
+        return makeTranslate([](Vector<Ref<CSSNumericValue>> components) {
+            ASSERT(components.size() == 3);
```

Comment:
check these, ASSERT_NOT_REACHED, return exception.

Reply:
Addressed in 6b8ab023c1ea.